code/graph_algorithms/src/centroid_decomposition/centroid_decompostition.hpp

```diff
@@ -0,0 +1,81 @@
+#include <iostream>
+#include <vector>
+
+class Graph 
+{
+	std::vector<int> empt;
+	int ver_;
+	std::vector<std::vector<int>> adj_;
+	std::vector<std::vector<int>> centroidTree_;
+	std::vector<int> sizes_;
+	std::vector<bool> marked_;
+    
+	Graph(int num) :
+        ver_(num),
+		adj_(num + 1),
+		centroidTree_(num + 1),
+		sizes_(num + 1),
+		marked_(num + 1)	
+    {}
+
+	void addEdge(int x, int y, std::vector<std::vector<int>> &graph);
+	void calcSizes(int cur, int par, std::vector<std::vector<int>> graph);
+	int findCentroid(int cur, int par, int vertices, std::vector<std::vector<int>> graph);
+	void decomposeTree(int cur, int par, int total, std::vector<std::vector<int>> graph);
+};
+
+void Graph::addEdge(int x, int y, std::vector<std::vector<int>> &graph)
+{
+    if (x < 0 || x > ver_)
+    {
+        throw std::out_of_range{"x is out of boundaries"};
+    }
+    if (y < 0 || y > ver_)
+    {
+        throw std::out_of_range{"y is out of boundaries"};
+    }
+	graph[x].push_back(y);
+}
+
+void Graph::calcSizes(int cur, int par, std::vector<std::vector<int>> graph)
+{
+	sizes_[cur] = 1;
+	for (const auto& to : graph[cur])
+	{
+		if (!(to == par || marked_[to] == true))
```

Comment:
No need for the `== true`, which is implicit.

Reply:
@arnavb I wrote it as such, so code would be more understandable (when you are fast skimming through the code this way it will be easily readable instead without "== true"), but if it is needed to be as such I will delete it

Reply:
@KtlTheBest I feel that when I read `|| marked_[to]`, the code is checking whether the edge (?) is marked, versus checking whether it is equal to true. It's like saying this this:

```C++
while (!gameOver)
```
is clearer than:
```C++
while (gameOver == false)
```